TechnicalDocuments/TechnicalDebt.md

```diff
@@ -0,0 +1,115 @@
+| ImageBusinessControllerProtocol | BabylonAppointmentsUI, BabylonClinicalRecordsUI, BabylonMonitor, Babylon |
+| BookAppointmentBusinessControllerProtocol | BabylonAppointmentsUI |
+| PrivacyBusinessControllerProtocol | BabylonAppointmentsUI, BabylonChatBotUI, BabylonUI |
+| PublicHealthcareIdentifierBusinessController | BabylonAppointmentsUI, BabylonClinicalRecordsUI, BabylonUI, Babylon |
+| SearchAssistantBusinessController | BabylonChatBotUI |
+| MedicalHistoryBusinessController | BabylonClinicalRecordsUI |
+| GenderBusinessControllerProtocol | BabylonClinicalRecordsUI, BabylonUI, Babylon |
+| GeolocationBusinessController | BabylonClinicalRecordsUI, BabylonUI |
+| InfoItemsBusinessControllerProtocol | BabylonClinicalRecordsUI |
+| MapsBusinessControllerProtocol | BabylonClinicalRecordsUI |
+| GPDetailsBusinessController | BabylonClinicalRecordsUI |
+| HealthcheckReportBusinessControllerProtocol | BabylonMonitor, BabylonUI, Babylon |
+| ReferAFriendBusinessControllerProtocol | BabylonUI, Babylon |
+| RatingBusinessController | BabylonUI |
+| FamilyBusinessControllerProtocol | BabylonUI |
+| RedemptionBusinessController | Babylon |
+| RegionBusinessControllerProtocol | Babylon |
+| EligibilityCheckBusinessControllerProtocol | Babylon |
+| NotificationPayloadBusinessControllerProtocol | Babylon |
+| SignInPrivacyBusinessControllerProtocol | Babylon |
+| PaymentsBusinessControllerProtocol | Babylon |
+| PDSQueueBusinessControllerProtocol | Babylon |
+| LocationEligibilityBusinessController | Babylon |
+| PostCodeEligibilityBusinessControllerProtocol | Babylon |
+| SignUpBusinessController | Babylon |
+| SignUpPrivacyBusinessController | Babylon |
+| ForgotPasswordBusinessControllerProtocol | Babylon |
+| FamilyBusinessControllerProtocol | Babylon |
+| OnfidoBusinessControllerProtocol | Babylon |
+| PDSBusinessControllerProtocol | Babylon |
+| NHSBusinessController | Babylon |
+| CreditCardsBusinessControllerProtocol | Babylon |
+| TestsAndKitsBusinessControllerProtocol | Babylon |
+| RedemptionBusinessControllerProtocol | Babylon |
+| ForgotPasswordBusinessController | Babylon|
+
+
+## Main Tab Bar and Navigation.
+
+Although much improved since version two was released, there is still a fair amount of confusing legacy code left in the tab bar flow controller and associated builders. Now that all content has been refactored to be based on reactive functional programming, flow controllers and builders it is possible to align `ExchangeAbleTabBarBuilder` and `TabContentBuilder` entities to follow our general design pattern. For historical reasons some flow controllers acts have logic for building view controllers. The `BabylonTabBarViewModel` should be able to forward all routing events to the appropriate flow controller. Some networking logic should be removed from `BabylonTabBarViewModel`. These deviations from our standard architecture makes it difficult to understand the code and changes often result in long and confusing discussions.
```

Comment:
```suggestion
Although much improved since version two was released, there is still a fair amount of confusing legacy code left in the tab bar flow controller and associated builders. Now that all content has been refactored to be based on reactive functional programming, flow controllers and builders it is possible to align `ExchangeableTabBarBuilder` and `TabContentBuilder` entities to follow our general design pattern. For historical reasons some flow controllers acts have logic for building view controllers. The `BabylonTabBarViewModel` should be able to forward all routing events to the appropriate flow controller. Some networking logic should be removed from `BabylonTabBarViewModel`. These deviations from our standard architecture makes it difficult to understand the code and changes often result in long and confusing discussions.
```